### PR TITLE
Post details

### LINKED ISSUE #6: Endpoint (GET) for post details
Should also return a list of the posts comments and info related to that user (who created the post) as well as the usernames and avatars for each commenter within that list

### PR DESCRIPTION
**Pull request by JoePeterson51**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Linter has passed locally and any fixes were made for failures
- [x] Travis build is passing
- [x] All tests passing locally

## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## What is the new behavior?

- Returns all the info for a post
- 
closes #6

### CODE REVIEW

File: spec/graphql/types/query_type/post_details_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    def post_query
+      <<~GQL
+        query {
+          post(id: "#{@posts1.id}") {
+            id
+            title
+            description
+            image
```
Comment (mkrumholz) [inline spec/graphql/types/query_type/post_details_spec.rb:48]: @JoePeterson51 FYI we probably also will want to test for sending up and down votes on both the post and comment level. I am guessing it works fine, but could be a nice add here to ensure we're getting that for the FE

File: spec/graphql/types/query_type/posts_spec.rb [hunk 2]
```diff
@@ -17,11 +17,11 @@
 
       json = JSON.parse(response.body, symbolize_names: true)
       data = json[:data][:posts]
-      
+
       expect(data).to be_an Array
       expect(data.length).to eq 10
 
-      expect(data.first[:id]).to eq "1"
+      expect(data.first[:id]).to eq "2"
```
Comment (mkrumholz) [inline spec/graphql/types/query_type/posts_spec.rb:24]: Was this supposed to change in this spec? Maybe we should put a user1.id or whatever into the spec instead if that was causing issues.

File: spec/graphql/types/query_type/post_details_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      expect(data[:description]).to eq(@posts1.description)
+      expect(data[:image]).to eq(@posts1.image)
+      expect(data[:user][:id].to_i).to eq(@user1.id)
+      expect(data[:user][:username]).to eq(@user1.username)
+      expect(data[:user][:avatar]).to eq(@user1.avatar)
+      expect(data[:comments].count).to eq(3)
+      expect(data[:comments][1][:content]).to eq(@comment2.content)
+      expect(data[:comments][1][:user][:id].to_i).to eq(@user2.id)
+      expect(data[:comments][1][:user][:username]).to eq(@user2.username)
+      expect(data[:comments][1][:user][:avatar]).to eq(@user2.avatar)
```
Comment (mkrumholz) [inline spec/graphql/types/query_type/post_details_spec.rb:38]: Nice! Like the thorough testing here.

File: app/graphql/types/post_type.rb [hunk 4]
```diff
@@ -10,5 +10,6 @@ class PostType < Types::BaseObject
     field :created_at, GraphQL::Types::ISO8601DateTime, null: false
     field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
     field :user, Types::UserType, null: false
+    field :comments, [Types::CommentType], null: true
```
Comment (mkrumholz) [inline app/graphql/types/post_type.rb:13]: Awesome work with this. Looks slick!

File: spec/graphql/types/query_type/post_details_spec.rb (hunk 1 above)
Comment (mkrumholz) [reply spec/graphql/types/query_type/post_details_spec.rb:48]: Confirmed that this does work when I pulled down the repo

**Review by mkrumholz — APPROVED** [review]
Looks awesome, Joe! Pulled down and tested and it's working exactly as I expect.

File: spec/graphql/types/query_type/posts_spec.rb (hunk 2 above)
Comment (mkrumholz) [reply spec/graphql/types/query_type/posts_spec.rb:24]: I updated this on my branch to be `user1.id.to_s` so it is situationally flexible -- you should do that on yours and push it up so the Travis build is good with any data probably

File: spec/graphql/types/query_type/post_details_spec.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+              upvotes
+              downvotes
+              user {
+                id
+                username
+                avatar
+              }
+            }
+          }
+        }
```
Comment (clairefields15) [inline spec/graphql/types/query_type/post_details_spec.rb:72]: looks great! Would love to see this query as a wiki like Molly did for all posts (whenever it is ready) :)

**Review by clairefields15 — APPROVED** [review]
Looks great to me!

**Pull request merged**